feat: link, breadcrumb, & button FC’s

alexsasharegan opened the pull request:
Requires the card PR that includes a util before moving forward.

tmorehouse commented on package.json:
```diff
@@ -56,7 +56,7 @@
     "docs-gen": "nuxt generate",
     "docs-publish": "./build/gh-pages",
-    "test": "jest --maxWorkers=1",
+    "test": "jest",
```
Maybe set this to 2

pi0 commented:
Something probably not related to this PR, but most Linters prefer single quotes why are we using such customized rules ? :))

alexsasharegan commented:
I think there's an open PR for linter settings. I just let my editor do the formatting for me, so dive into the eslint stuff and set it up however you'd like!

pi0 commented:
Will locally check #792 soon :)

alexsasharegan commented:
I'm pretty happy we (@tmorehouse & I) got prop sync working on a functional component. Could be wrong, but we might literally be the first to do that.

alexsasharegan commented:
Fixed a couple components referencing `button.vue` directly, which led me to mess with modal. I fixed a lot of spelling errors in the docs there, and a quirky bug in the demo. Basically, you could keep submitting names without the modal closing by using the enter key. I fixed that, and in the process show off a couple ways to work with modal events.

tmorehouse commented on docs/components/modal/README.md:
```diff
@@ -71,19 +76,19 @@ via the `modal-footer` slot. Note when using the `modal-footer` slot the default
 
-#### Using `v-b-modal` directive (recommended)
+### Using `v-b-modal` directive (recommended)
 
```
This should be 4 hashes, or change th ones below to 3 hashes, since the headings are at the same sub level under Toggle Modal Visibility

alexsasharegan replied:
The whole thing was freaking out about not descending H's in proper order.

mosinve replied:
Yep, H's levels have to be changes

alexsasharegan commented:
Okay, I've gone through the docs page-by-page. I've cleared all the console errors I could find, and tried to get any spelling typos along the way.

mosinve commented:
Yep, i think it's finished ))

alexsasharegan commented:
Cool, once @tmorehouse approves, I'll squash & merge.

mosinve commented:
+1,380 −1,677 - lib going to be slightly slimmer ))))

alexsasharegan commented:
Haha but it's mostly deleting all the old jsfiddle's!

tmorehouse commented on docs/components/pagination-nav/README.md:
```diff
@@ -10,11 +10,11 @@ regular links or router links.
 
     <h6 class="mt-4">With link generator function</h6>
-    <b-pagination-nav link-gen="linkGen" :number-of-pages="10" v-model="currentPage" />
+    <b-pagination-nav :link-gen="linkGen" :number-of-pages="10" v-model="currentPage" />
```
👍
